presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java

```diff
@@ -5173,6 +5173,61 @@ public void testLateralJoin()
         assertQuery(
                 "SELECT * FROM (VALUES 1, 2) a(x) JOIN LATERAL(SELECT y FROM (VALUES 2, 3) b(y) WHERE y > x) c(z) ON z > 2*x",
                 "VALUES (1, 3)");
+
+        // Limit in correlated subquery
+        assertQuery(
+                "SELECT * " +
+                        "FROM (VALUES 1, 2, 3) outer_relation(id) " +
+                        "LEFT JOIN LATERAL " +
+                        "(SELECT f1 FROM " +
+                        "(VALUES " +
+                        "(1, 'a'), " +
+                        "(1, 'a'), " +
+                        "(1, 'a'), " +
+                        "(1, 'a'), " +
+                        "(2, 'b')) inner_relation(f0, f1) " +
+                        "WHERE outer_relation.id = inner_relation.f0 " +
+                        "LIMIT 2) " +
+                        "ON TRUE",
+                "VALUES " +
+                        "(1, 'a'), " +
+                        "(1, 'a'), " +
+                        "(2, 'b'), " +
+                        "(3, null)");
+
+        // TopN in correlated subquery
+        assertQuery(
+                "SELECT regionkey, n.name FROM region LEFT JOIN LATERAL (SELECT name FROM nation WHERE region.regionkey = regionkey ORDER BY nationkey LIMIT 2) n ON TRUE",
+                "VALUES " +
+                        "(0, 'ETHIOPIA'), " +
+                        "(0, 'ALGERIA'), " +
+                        "(1, 'BRAZIL'), " +
+                        "(1, 'ARGENTINA'), " +
+                        "(2, 'INDONESIA'), " +
+                        "(2, 'INDIA'), " +
+                        "(3, 'GERMANY'), " +
+                        "(3, 'FRANCE'), " +
+                        "(4, 'IRAN'), " +
+                        "(4, 'EGYPT')");
+        assertQuery(
+                "SELECT * " +
+                        "FROM (VALUES 1, 2, 3) outer_relation(id) " +
+                        "LEFT JOIN LATERAL " +
+                        "(SELECT f1 FROM " +
+                        "(VALUES " +
+                        "(1, 'd'), " +
+                        "(1, 'c'), " +
+                        "(1, 'b'), " +
+                        "(1, 'a'), " +
+                        "(2, 'w')) inner_relation(f0, f1) " +
+                        "WHERE outer_relation.id = inner_relation.f0 " +
+                        "ORDER BY inner_relation.f1 LIMIT 2) " +
+                        "ON TRUE",
+                "VALUES " +
+                        "(1, 'b'), " +
```

Comment:
Order is not respected after a join and so `b` is before `a`, so this test could be flaky as it possible to get different results. Please add `ORDER BY 1,2` at the end of your query.

Reply:
No, `assertQuery()` compares the results ignoring order, which is actually a must for testing any query without `ORDER BY` at top level.